src/frontends/onnx/frontend/src/op/layer_normalization.cpp

```diff
@@ -6,95 +6,81 @@
+    int64_t stash_type_i = node.get_attribute_value<int64_t>("stash_type", default_stash_type_i);
     element::Type stash_type = common::get_ov_element_type(stash_type_i);
 
-    // following calculations are kept as close to the onnx\defs.cc description as possible
-    auto FloatEpsilon = Constant::create(ov::element::f32, Shape{}, {epsilon});
-    auto Epsilon = std::make_shared<Convert>(FloatEpsilon, stash_type);
-    auto XShape = std::make_shared<ShapeOf>(X);
-    auto Rank = std::make_shared<v3::ShapeOf>(XShape);
-    auto Zero1D = Constant::create(ov::element::i64, {1}, {0});
-    auto One1D = Constant::create(ov::element::i64, {1}, {1});
-    auto Axis1D = Constant::create(ov::element::i64, {1}, {axis});
-    auto PrefixShape = std::make_shared<Slice>(XShape, Zero1D, Axis1D, One1D);
-    ov::Output<ov::Node> NumReducedAxes = (axis >= 0 ? std::make_shared<Subtract>(Rank, Axis1D)->output(0)
-                                                     : std::make_shared<Negative>(Axis1D)->output(0));
-    auto SuffixShape = std::make_shared<v3::Broadcast>(One1D, NumReducedAxes);
-    auto ReducedShape = std::make_shared<Concat>(ov::OutputVector{PrefixShape, SuffixShape}, 0);
+    ov::Output<ov::Node> data = inputs.at(0);
+    element::Type original_type = data.get_element_type();
+    bool needs_type_casting = stash_type != original_type;
 
-    auto X2D = ov::op::util::flatten(X, static_cast<int>(axis));
-    auto XU = std::make_shared<Convert>(X2D, stash_type);
+    if (needs_type_casting)
+        data = std::make_shared<Convert>(data, stash_type);
 
-    auto Mean2D = std::make_shared<ReduceMean>(XU, One1D, true);
-    auto Square = std::make_shared<Multiply>(XU, XU);
-    auto MeanOfSquare = std::make_shared<ReduceMean>(Square, One1D, true);
-    auto SquareOfMean = std::make_shared<Multiply>(Mean2D, Mean2D);
+    float epsilon = node.get_attribute_value<float>("epsilon", 1e-5f);
 
-    auto Var = std::make_shared<Subtract>(MeanOfSquare, SquareOfMean);
-    auto VarPlusEpsilon = std::make_shared<Add>(Var, Epsilon);
-    auto StdDev = std::make_shared<Sqrt>(VarPlusEpsilon);
-    auto Deviation = std::make_shared<Subtract>(XU, Mean2D);
-    auto Normalized = std::make_shared<Divide>(Deviation, StdDev);
-    auto NormalizedT = std::make_shared<ConvertLike>(Normalized, X);
+    auto axis = node.get_attribute_value<std::int64_t>("axis", -1);
+    // ONNX operator semantics says that `axis` attribute points to the first normalization dimension. We have to
+    // figure out all dimensions for normalization:
+    // axis < 0:  range(axis, 0)     Example: axis = -2; Axes: [-2, -1]
+    // axis >= 0: range(axis, rank)  Example: axis = 2, rank = 4; Axes: [2, 3]
+    auto axes =
+        std::make_shared<v4::Range>(Constant::create(element::i64, {}, {axis}),
+                                    (axis < 0 ? Constant::create(element::i64, {}, {0})->output(0) : rank(data)),
```

Comment:
Is there a specific reason to use `->output(0)` for the Constant here?

Reply:
due to trernary operator